packages/editor/src/components/document-bar/index.js

```diff
@@ -48,51 +48,32 @@ const icons = {
 };
 
 export default function DocumentBar() {
-	const {
-		isEditingTemplate,
-		templateId,
-		postType,
-		postId,
-		goBack,
-		getEditorSettings,
-	} = useSelect( ( select ) => {
+	const { postType, postId, goBack } = useSelect( ( select ) => {
 		const {
-			getRenderingMode,
-			getCurrentTemplateId,
 			getCurrentPostId,
 			getCurrentPostType,
 			getEditorSettings: getSettings,
 		} = select( editorStore );
-		const _templateId = getCurrentTemplateId();
 		const back = getSettings().goBack;
 		return {
-			isEditingTemplate:
-				!! _templateId && getRenderingMode() === 'template-only',
-			templateId: _templateId,
 			postType: getCurrentPostType(),
 			postId: getCurrentPostId(),
 			goBack: typeof back === 'function' ? back : undefined,
 			getEditorSettings: getSettings,
 		};
 	}, [] );
 
-	const { setRenderingMode } = useDispatch( editorStore );
-
 	const handleOnBack = () => {
-		if ( isEditingTemplate ) {
-			setRenderingMode( getEditorSettings().defaultRenderingMode );
-			return;
-		}
 		if ( goBack ) {
 			goBack();
 		}
```

Comment:
I wonder if the issue might be the code deleted on line 82-85, which seems important and maybe shouldn't be deleted. The `defaultRenderingMode` is one of `template-locked` or `all`, so not related to `template-only`.

Also noticing that currently when you edit a page, all blocks seems to be editable, compared to in trunk where only the post content is editable.

Reply:
> I wonder if the issue might be the code deleted on line 82-85, which seems important and maybe shouldn't be deleted. 

Because we are loading a new entity into the editor now on going back the default rendering mode will be set on the new editor load so I don't think we also need to do it here.

>Also noticing that currently when you edit a page, all blocks seems to be editable, compared to in trunk where only the post content is editable.

This was caused by the way I was setting the default rendering mode - should be fixed now.

Reply:
Yes, the goBack method shouldn't passed to the site editor provider unless we actually went into focus mode, so the site editor need to be a second history of focused entities (in addition to the url history) kind of similar to the post editor. And navigating in the site editor outside these methods should reset the focused entities history.

Reply:
> This was caused by the way I was setting the default rendering mode - should be fixed now.

Hmm, still doesn't seem to be right, the page content doesn't show 🤔

Reply:
@youknowriad Could an alternative be passing `goBack` in as a prop to any `editor` components?
```jsx
<DocumentBar goBack={ goBack } />
```

That seems like it would circumvent any need for a second history and goBack implementation in the site editor.

Reply:
for me `goBack` just means `isFocusMode` which translates to:

 - Having a callback to go back to the previous focus entity or parent entity.
 - Showing a padding around the canvas.

So we need in two places.

The other thing is, I'd like to arrive to a point in the next months where we could render the whole editor using `<Editor post={post} settings={settings) />` to render the whole editor. You won't use the editor components one by one, which means at some point we need to address these specific components passed to specific components (probably become settings in most cases if really needed or just go away). That said, adding props to these components is fine temporarily (especially if these components are private APIs)

Reply:
> Hmm, still doesn't seem to be right, the page content doesn't show 🤔

This is hopefully fixed [by this](https://github.com/WordPress/gutenberg/pull/57700/commits/a749611e1f3c2e1a48b5c1ce98b0fab7853dae38) @talldan

Reply:
Adding a goBack option in the site editor is a little more complex because it has a router and the back button works. At a quick glance it looks like we would need to add something like a `popstate` eventlistener and try and work out if we are coming back from a focus mode page in order to clear the goBack setting. 

There may be an easier way, but I would like to suggest we keep it simple for this PR and just use the existing `focusMode` param that is used already in the site editor to get the back link (using the router back method) and the editor padding.  [This change ](https://github.com/WordPress/gutenberg/pull/57700/commits/d6bebab08dc6dda6d6f22ab5b3652d2b945dd86a)does this and it seems to work, and we can always revisit this in a follow up if we do want to tie it in closer with the router.

Reply:
While it's a follow-up to a follow-up, making it another follow-up seems ok to me. This PR is getting quite big and comment-y, so breaking it out into a separate PR would make it easier to review.

Reply:
yeh, isFocusMode has been in the code base for at least 6 months, and this PR is intended to remove the recently added `template-only` mode before 6.5, so I think it makes sense rather than doing a follow-up directly after this to instead look at removing `isFocusedMode` as a separate task post 6.5, and which point we will have more time to plan and test more complex routing changes.